server/migrator/Schema.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    private void writeRelationType(StringBuilder builder, RelationType relationType) {
+        builder.append(String.format("%s sub %s",
+                relationType.getLabel().name(),
+                relationType.getSupertype().getLabel().name()));
+        writeOwns(builder, relationType);
+        writeRelates(builder, relationType);
+        writePlays(builder, relationType);
+        builder.append(SEMICOLON_NEWLINE_X2);
+    }
+
+    private void writeEntityType(StringBuilder builder, EntityType entityType) {
+        builder.append(String.format("%s sub %s",
+                entityType.getLabel().name(),
+                entityType.getSupertype().getLabel().name()));
+        writeOwns(builder, entityType);
+        writePlays(builder, entityType);
+        builder.append(SEMICOLON_NEWLINE_X2);
+    }
+
+    private void writeOwns(StringBuilder builder, ThingType thingType) {
+        Set<String> keys = thingType.getOwns(true).map(x -> x.getLabel().name()).collect(Collectors.toSet());
+        List<String> attributeTypes = thingType.getOwns().map(x -> x.getLabel().name()).collect(Collectors.toList());
+        attributeTypes.stream().filter(x -> keys.contains(x)).sorted()
+                .forEach(x -> builder
+                        .append(COMMA_NEWLINE_INDENT)
+                        .append(String.format("owns %s @key", x)));
+        attributeTypes.stream().filter(type -> !keys.contains(type)).sorted()
+                .forEach(x -> builder
+                        .append(COMMA_NEWLINE_INDENT)
+                        .append(String.format("owns %s", x)));
+    }
+
+    private void writeRelates(StringBuilder builder, RelationType relationType) {
+        relationType.getRelates().sorted(Comparator.comparing(x -> x.getLabel().name()))
+                .forEach(x -> builder
+                        .append(COMMA_NEWLINE_INDENT)
+                        .append(String.format("relates %s", x.getLabel().name())));
+    }
+
+    private void writePlays(StringBuilder builder, ThingType thingType) {
+        thingType.getPlays().sorted(Comparator.comparing(x -> x.getLabel().scopedName()))
+                .forEach(x -> builder
+                        .append(COMMA_NEWLINE_INDENT)
+                        .append(String.format("plays %s", x.getLabel().scopedName())));
+    }
+
+    private void writeRule(StringBuilder builder, Rule rule) {
+        builder.append(String.format("%s sub rule", rule.getLabel()))
+                .append(COMMA_NEWLINE_INDENT)
+                .append("when\n")
+                .append(getPatternString(wrapConjunction(rule.getWhenPreNormalised()), 1))
+                .append(COMMA_NEWLINE_INDENT)
+                .append("then\n")
+                .append(getPatternString(wrapConjunction(rule.getThenPreNormalised()), 1))
+                .append(SEMICOLON_NEWLINE_X2);
+    }
+
+    private String getValueTypeString(AttributeType.ValueType valueType) {
+        switch (valueType) {
```

Comment:
I think it's fine to replace this method with
```java
return valueType.name().toLowerCase();
```
If you want to add a safety net you can add
`if (valueType == AttributeType.ValueType.OBJECT) throw GraknException.of(ILLEGAL_STATE);` but I think that's optional.

Reply:
Emm... I'd rather not to do that, I don't want to rely on the enum names. What happens if you rename `DATETIME` to `DATE_TIME`.

Reply:
Fair enough